go/kbnm/handler.go

```diff
@@ -53,25 +59,99 @@ func (h *handler) handleChat(req *Request) error {
 		return errMissingField
 	}
 
-	binPath, err := findKeybaseBinary()
+	binPath, err := h.FindKeybaseBinary()
 	if err != nil {
 		return err
 	}
 
+	var out bytes.Buffer
 	cmd := exec.Command(binPath, "chat", "send", "--private", req.To)
+	cmd.Env = append(os.Environ(), "KEYBASE_LOG_FORMAT=plain")
 	cmd.Stdin = strings.NewReader(req.Body)
+	cmd.Stdout = &out
+	cmd.Stderr = &out
 
-	// TODO: Check/convert status code more precisely? Maybe return stdout as
-	// part of the error if there is one?
-	err = h.Run(cmd)
+	if err := h.Run(cmd); err != nil {
+		return parseError(&out, err)
+	}
 
-	return err
+	return nil
 }
 
 type resultQuery struct {
-	Sigs []struct {
-		Statement string `json:"statement"`
-	} `json:"sigs"`
+	Username string `json:"username"`
+}
+
+// parseQuery reads the stderr from a keybase query command and returns a result
+func parseQuery(r io.Reader) (*resultQuery, error) {
+	scanner := bufio.NewScanner(r)
+
+	for scanner.Scan() {
+		// Find a line that looks like... "[INFO] 001 Identifying someuser"
+		line := strings.TrimSpace(scanner.Text())
+		parts := strings.Split(line, " ")
+		if len(parts) < 4 {
+			continue
+		}
+
+		if parts[2] != "Identifying" {
+			continue
+		}
```

Comment:
Is it bad that this could also match a line like `[ERRO] 123 Identifying someuser failed!`